src/web/handlers.rs

```diff
@@ -61,18 +65,62 @@ pub async fn get_tiles(
         region_code: location.region(),
         form_factor: device_info.form_factor,
     };
+
     let mut expired = false;
     if !settings.test_mode {
-        if let Some(tiles) = state.tiles_cache.get(&audience_key) {
-            expired = tiles.expired();
-            if !expired {
-                trace!("get_tiles: cache hit: {:?}", audience_key);
-                metrics.incr("tiles_cache.hit");
-                return Ok(content_response(&tiles.content));
+        // First make a cheap read from the cache
+        if let Some(tiles_state) = state.tiles_cache.get(&audience_key) {
+            match &*tiles_state {
+                TilesState::Populating => {
+                    // Another task is currently populating this entry and will
+                    // complete shortly. 503 until then instead of queueing
+                    // more redundant requests
+                    trace!("get_tiles: Another task Populating");
+                    metrics.incr("tiles_cache.miss.populating");
+                    return Ok(HttpResponse::ServiceUnavailable().finish());
```

Comment:
I've begun some work on making this case instead "watch" for completion via a tokio watch channel. I'm not yet sure it's worth it (if this will even be that significant number of 503s)

Reply:
I was gonna comment on this `return-503-if-populating`. If we end up seeing a high volume of 503s, Firefox might miss some impressions&clicks. We might want to let those requests "spin" a bit in hope that the population could be done soon.

Without any instrumentation, I am inclined to treat that as a potential follow-up before further complicating this patch.